src/hardware/sblaster.cpp

```diff
@@ -2358,18 +2358,21 @@ class SBLASTER final {
 
 }; // End of SBLASTER class
 
-static std::unique_ptr<SBLASTER> sblaster = {};
-
-void SBLASTER_ShutDown(Section* /*sec*/) {
-	sblaster = {};
-}
-
 void SBLASTER_Init(Section* sec)
 {
-	assert(sec);
+	static std::unique_ptr<SBLASTER> sblaster_instance = {};
 
-	sblaster = std::make_unique<SBLASTER>(sec);
+	// Always reset the Sound Blaster before configuring
+	sblaster_instance = {};
 
-	constexpr auto changeable_at_runtime = true;
-	sec->AddDestroyFunction(&SBLASTER_ShutDown, changeable_at_runtime);
+	if (sec) {
+		sblaster_instance = std::make_unique<SBLASTER>(sec);
+	}
+}
+
+void SBLASTER_ShutDown(Section*)
+{
+	// shutdown with an empty section
+	constexpr Section* shutdown = nullptr;
+	SBLASTER_Init(shutdown);
```

Comment:
Uh, that's a bit weird. Can't we avoid that? Maybe extracting the common logic out from `SBLASTER_Init` then calling it from both Init and Shutdown?